gulpfile.js

```diff
@@ -216,6 +216,8 @@ gulp.task('deploy-data-programs', [ 'deploy-clean' ], function () {
                 steps: data.path.reportedSpeed, // data.averageSteps, // @todo until step measurement matches game's
                 successRatio: data.successRatio,
                 type: data.path.type,
+                legal: ( (!/(exploit|specific|obsolete)/.exec(data.path.type)) && ((data.successRatio == 1) || ([4, 28, 41].includes(data.level.number))) ),
```

Comment:
1. `.test()` can be used instead of `.exec()`
2. I prefer to use `===` over `==`